networks/known_networks_test.go

```diff
@@ -220,27 +220,325 @@ func TestNewEVMNetwork(t *testing.T) {
+selected_networks = ["new_network"]
+
+[EVMNetworks.new_network]
+evm_name = "new_test_network"
+evm_chain_id = 100009
+evm_urls = ["ws://localhost:8546"]
+evm_http_urls = ["http://localhost:8545"]
+evm_keys = ["ac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80"]
+evm_simulated = true
+evm_chainlink_transaction_limit = 5000
+evm_minimum_confirmations = 1
+evm_gas_estimation_buffer = 10000
+client_implementation = "Ethereum"
+evm_supports_eip1559 = true
+evm_default_gas_limit = 6000000
+`,
+			expNetworks: []blockchain.EVMNetwork{newNetwork},
+		},
+		{
+			name: "new network with rpc urls in EVMNetworks and wallet keys in WalletKeys NetworkConfig",
+			networkConfigTOML: `
+selected_networks = ["new_network"]
+
+[EVMNetworks.new_network]
+evm_name = "new_test_network"
+evm_chain_id = 100009
+evm_urls = ["ws://localhost:8546"]
+evm_http_urls = ["http://localhost:8545"]
+evm_simulated = true
+evm_chainlink_transaction_limit = 5000
+evm_minimum_confirmations = 1
+evm_gas_estimation_buffer = 10000
+client_implementation = "Ethereum"
+evm_supports_eip1559 = true
+evm_default_gas_limit = 6000000
+
+[WalletKeys]
+new_network = ["ac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80"]
+`,
+			expNetworks: []blockchain.EVMNetwork{newNetwork},
+		},
+		{
+			name: "new network with rpc urls and wallet keys in NetworkConfig",
+			networkConfigTOML: `
+selected_networks = ["new_network"]
+
+[EVMNetworks.new_network]
+evm_name = "new_test_network"
+evm_chain_id = 100009
+evm_simulated = true
+evm_chainlink_transaction_limit = 5000
+evm_minimum_confirmations = 1
+evm_gas_estimation_buffer = 10000
+client_implementation = "Ethereum"
+evm_supports_eip1559 = true
+evm_default_gas_limit = 6000000
+
+[RpcHttpUrls]
+new_network = ["http://localhost:8545"]
+[RpcWsUrls]
+new_network = ["ws://localhost:8546"]
+[WalletKeys]
+new_network = ["ac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80"]
+`,
+			expNetworks: []blockchain.EVMNetwork{newNetwork},
+		},
+	}
+	for _, tc := range testcases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			encoded := base64.StdEncoding.EncodeToString([]byte(tc.networkConfigTOML))
+			err := os.Setenv("BASE64_NETWORK_CONFIG", encoded)
+			require.NoError(t, err, "error setting env var")
+
+			networkCfg := &config.NetworkConfig{}
+			networkCfg.UpperCaseNetworkNames()
+			err = networkCfg.Default()
+			require.NoError(t, err, "error setting default network config")
+			err = networkCfg.Validate()
+			if tc.isNetworkConfigError {
+				require.Error(t, err, "expected network config error")
+				return
+			}
+			require.NoError(t, err, "error validating network config")
+			actualNets, err := SetNetworks(*networkCfg)
+			if tc.isEVMNetworkError {
+				t.Log(err)
+				require.Error(t, err, "expected evmNetwork set up error")
+				return
+			}
+			require.NoError(t, err, "unexpected error")
+			require.Equal(t, tc.expNetworks, actualNets, "unexpected networks")
+		})
+	}
+}
```

Comment:
nice job! I think i'm only missing some tests for forkconfig/evm network in defaults? I saw you added that functionality